qiskit_nature/circuit/library/ansatzes/ucc.py

```diff
@@ -123,6 +123,8 @@ def __init__(
         alpha_spin: bool = True,
         beta_spin: bool = True,
         max_spin_excitation: Optional[int] = None,
+        generalized: bool = False,
+        spin_flip: bool = False,
```

Comment:
I am just wondering about this as a name `spin_flip`. Generalized seems more like setting the strategy allowed when building things out. `spin_flip` sounds more like something it will do, as in change all the spins. How does `preserve_spin` seem (defaults True) - in that excitations will preserve whatever spins the particles have and will not change them i.e. it will not excite from alpha to beta or vice versa.